src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -405,6 +407,16 @@ int sd_dhcp_client_set_vendor_class_identifier(
         return free_and_strdup(&client->vendor_class_identifier, vci);
 }
 
+
+int sd_dhcp_client_set_user_class(
+                sd_dhcp_client *client,
+                const char *user_class) {
```

Comment:
Ok then in this special case we should introduce a specific API fir user class  `int dhcp_option_append_user_class(DHCPMessage *message, size_t size, size_t *offset, const char **optval) `  which should handle `char **`  . Changing signature `dhcp_option_append` will break the existing API and lead to lots of code change.